olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -378,6 +378,65 @@ client::CancellableFuture<PollResponse> StreamLayerClientImpl::Poll() {
                                                         std::move(promise));
 }
 
+client::CancellationToken StreamLayerClientImpl::Seek(
+    SeekRequest request, SeekResponseCallback callback) {
+  auto seek_task = [=](client::CancellationContext context) -> SeekResponse {
+    std::string subscription_id;
+    std::string subscription_mode;
+    std::string x_correlation_id;
+    std::shared_ptr<client::OlpClient> client;
+
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      if (!client_context_) {
+        OLP_SDK_LOG_WARNING_F(kLogTag, "Seek: unsuccessful, not subscribed");
+        return client::ApiError(client::ErrorCode::PreconditionFailed,
+                                "Not subscribed", false);
+      }
+
+      subscription_id = client_context_->subscription_id;
+      subscription_mode = client_context_->subscription_mode;
+      x_correlation_id = client_context_->x_correlation_id;
+      client = client_context_->client;
+    }
+
+    auto const& offsets = request.GetOffsets();
```

Comment:
This method uses only the `offsets` from the `request`, so would it be better all `Seek` methods asked for offsets instead of the entire request object?

Reply:
Good point.
I was following stream layer design docs. Are we planning to extend seek functionality, so the use of SeekRequest object is justified?

Reply:
Not sure. I planned it like this so that we can support BillingTag if needed. Not sure if it is even planned. The stream layer is still in "construction" and not broadly used so far. We might need some adjustments in the future.

Reply:
Sounds reasonable. Let's keep the request object.